app/src/main/java/com/hfad/mymessenger/ReceiveMessageActivity.kt

```diff
@@ -20,7 +20,12 @@ class ReceiveMessageActivity : AppCompatActivity() {
         var message = intent?.extras?.getString(CreateMessageActivity.MESSAGE).toString()
 
         //TODO: Add an exclamation mark to the TextView if the intent boolean data "URGENT" is true
-
+        var urgent = intent?.extras?.getBoolean(CreateMessageActivity.URGENT)?:false
+        if (urgent){
+            binding.message.text = message.uppercase() + "!"
+        }else{
+            binding.message.text = message
+        }
```

Comment:
Remove line 29 otherwise  `binding.message.text` will ALWAYS end up as the plain message